### PR TITLE
Feature/fine uploader

### PR DESCRIPTION
**Pull request by bradfordwagner**
Please fill in this template.

- [x] Make your PR against the `master` branch.
- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped#common-mistakes).
- [x] Run `tsc` without errors.
- [x] Run `npm run lint package-name` if a `tslint.json` is present.

Select one of these and delete the others:

If adding a new definition:
- [x] The package does not provide its own types, and you can not add them.
- [x] If this is for an NPM package, match the name. If not, do not conflict with the name of an NPM package.
- [ ] Create it with `npm run new-package package-name`, not by basing it on an existing project.

### CODE REVIEW

**Comment by andy-ms** [comment]
#13760 apparently hasn't kicked in yet. This needs a `tslint.json` containing `{ "extends": "../tslint" }`.

**Comment by andy-ms** [comment]
Sorry, that was my typo. Should be `"../tslint.json"`.

**Comment by andy-ms** [comment]
Seems to compile fine in `typescript@2.0`, why did you add `// TypeScript Version: 2.1`?

**Comment by bradfordwagner** [comment]
I am trying to fix: 
1:25  dt-header  Error parsing header. Expected: foo MAJOR.MINOR (patch version not allowed)

I'm not certain what causes it exactly. Is there any documentation that I missed?

This shouldn't require even 2.X. I"ll remove the version.

File: fine-uploader/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,337 @@
...
+        params?: P;                 // default: {}
+        paramsInBody?: boolean;     // default: true
+        uuid?: string;              // default: qquuid
+        totalFileSizeName?: string; // default: qqtotalfilesize
+    }
+
+    interface ScalingOptions {
+        customResizer?: (blob: File | Blob, height: number, image: HTMLImageElement, sourceCanvas: HTMLCanvasElement, targetCanvas: HTMLCanvasElement, width: number) => Promise<File | Blob>; // default: undefined
+        defaultQuality?: number;    // default: 80
+        defaultType?: string;       // default: null
```
Comment (mhegazy) [inline fine-uploader/index.d.ts]: should not thins be `string | null` if the default is `null`?

File: fine-uploader/index.d.ts [hunk 2]
```diff
@@ -0,0 +1,337 @@
...
+     */
+    interface Size {
+        name: string;
+        maxSize: number;
+        type?: string;
+    }
+
+    interface SessionOptions<H, P> {
+        customHeaders?: H;          // default: {}
+        endpoint?: string;          // default: null
```
Comment (mhegazy) [inline fine-uploader/index.d.ts]: and this.

File: fine-uploader/index.d.ts [hunk 3]
```diff
@@ -0,0 +1,337 @@
...
+    }
+
+    interface TextOptions {
+        defaultResponseError?: string;  // default: Upload failure reason unknown
+        fileInputTitle?: string;        // default: file input
+        sizeSymbols?: string[];         // default: ['kB', 'MB', 'GB', 'TB', 'PB', 'EB']
+    }
+
+    interface ValidationOptions {
+        acceptFiles?: MimeType[];           // default: null
```
Comment (mhegazy) [inline fine-uploader/index.d.ts]: and this.

File: fine-uploader/index.d.ts (hunk 1 above)
Comment (bradfordwagner) [inline fine-uploader/index.d.ts]: You're correct. I've made the appropriate alterations.

File: fine-uploader/index.d.ts (hunk 2 above)
Comment (bradfordwagner) [inline fine-uploader/index.d.ts]: fixed.

File: fine-uploader/index.d.ts (hunk 3 above)
Comment (bradfordwagner) [inline fine-uploader/index.d.ts]: fixed.

File: fine-uploader/index.d.ts [hunk 4]
```diff
@@ -0,0 +1,337 @@
+// Type definitions for FineUploader for 5.11.10
```
Comment (andy-ms) [inline fine-uploader/index.d.ts]: Lint error is here: `1:25 dt-header Error parsing header. Expected: foo MAJOR.MINOR (patch version not allowed)`
This line should be just `// Type definitions for FineUploader for 5.11`.
Then remove the last line (`// TypeScript Version: 2.1`).

**Comment by andy-ms** [comment]
Thanks!

**Comment by rnicholus** [comment]
This just popped onto my radar a few days ago. FYI a similar effort has been underway to bundle types inside of the package itself at FineUploader/fine-uploader#1689.

**Comment by andy-ms** [comment]
Bundled types are preferred, so we'll deprecate this once you've bundled.
However, your pull request doesn't [add `"types"` to `package.json`](http://www.typescriptlang.org/docs/handbook/declaration-files/publishing.html), so we wouldn't detect them as properly bundled.

**Comment by rnicholus** [comment]
I've asked the author of that PR to take a peek at this one, so I suspect that will be resolved before merging the PR. Thanks for the quick response.

**Comment by SinghSukhdeep** [comment]
Having some issues with the bundled types right now for the fine-uploader project.
I've added the `"types"` property declaration in `package.json`, at FineUploader/fine-uploader#1689. but TS isn't picking the def file.
I'm working with an Angular 2 based project to test this.

I can prepare a sample repo if someone can take a look and help

Works only if I place the def file under `node_modules/@types/fine-uploader/index.d.ts`.

**Comment by rnicholus** [comment]
Seems like this can/should be removed as a TS def file is now bundled with Fine Uploader as of [v5.14.0](https://github.com/FineUploader/fine-uploader/releases/tag/5.14.0).

**Comment by andy-ms** [comment]
@SinghSukhdeep You'll need `/// <reference types="fine-uploader" />` as types in `node_modules` aren't included automatically.
@rnicholus #14983

**Comment by rnicholus** [comment]
We have that listed in the ["test" file](https://github.com/FineUploader/fine-uploader/blob/5.14.0/client/typescript/fine-uploader.test.ts) which serves as an example. Documentation will need to be updated on our end. Thanks @andy-ms.

**Pull request merged**